### ISSUE TITLE
Running Stata code from VSCode

### ISSUE THREAD

**Issue by hozdenoren** [kylebarron/stata_kernel#397]
Hello,
Now that the VSCode supports Jupyter kernels (at least in the [insider's edition](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter)), I was wondering if it would be possible to leverage it to use Stata in VSCode via stata_kernel as we do in Atom? 
H

**Comment by amichuda** [comment]
That's already been working for a while now. At least on my experience, you just need to set the stata kernel and everything works pretty straightforwardly.

**Comment by kylebarron** [comment]
Last time I looked, Vscode really only supported Python kernels, and it only supported a cell interface, in contrast to a line-based interface like Hydrogen, so I never used it. It would be great to have a pull request to add documentation about how to use in Vscode.

**Comment by hozdenoren** [comment]
@amichuda  Do you mean it is working in Jupyter Notebooks within Vscode? I was wondering if it could also work like Atom with directly using do-files, getting in-line results and plots pane.

@kylebarron The insiders build supports other language kernels than Python.  Based on @amichuda's response, I think this also includes stata_kernel as well. And, I agree it would be nice to have it in the docs. For example, as of now, I am having issues getting it to work within Vscode, even though it works with JupyterLab-- but there is nowhere to look for help.

**Comment by amichuda** [comment]
@hozdenoren if you mean, whether it will work like Hydrogen, then no. it won't inline. But it does work within the standard `vscode-jupyter` experience, where you get a jupyter notebook interface. There's also the option of using the interactive window, so you can use scripts directly, but it's not implemented for non-python languages, although there is a pull request open to allow for code-cells on a per-language basis (https://github.com/microsoft/vscode-jupyter/pull/4876), but it still hasn't been merged.

Does that make sense? If you're having trouble having it work, shoot me a message. If it seems like it would be useful, I can write up a short doc

**Comment by kylebarron** [comment]
I think you could make a `vscode.md` doc here: https://github.com/kylebarron/stata_kernel/tree/master/docs/src/using_jupyter

**Comment by hozdenoren** [comment]
@amichuda Yes, that makes sense! Thanks for the responses! I will message you the issue that I have, not to clutter here. Maybe it could be useful for writing the doc.

**Comment by StevenLi-DS** [comment]
Hi, @kylebarron Is this anywhere to set the output font size in a jupyter env? I am using vscode now and something the results were wrapped and not very readable. It seems like all long results are wrapped and no horizontal scroll can not be triggered, which  is something has been happening in Jupyter Lab as well.

**Comment by StevenLi-DS** [comment]
@kylebarron Is there a known issue. It seems like plots won't show up in vscode:
![image](https://user-images.githubusercontent.com/33796896/126210121-caffbd2e-b8b9-40e2-b9c0-6eb67395594e.png)

**Comment by kylebarron** [comment]
You're welcome to look into it and make a Pull Request.

**Comment by StevenLi-DS** [comment]
@kylebarron seems like this is a Stata 17 related issue https://github.com/kylebarron/stata_kernel/pull/394 Sorry I forgot to mention I use Stata 17.

**Comment by hozdenoren** [comment]
Graphs work for me with Stata17 after I updated with the files in the PR.

**Comment by ledwindra** [comment]
Hi everyone,

I don't know if anyone wants to see this but it appears that we not only can execute Stata from VSCode, we can also run it on GitHub remote server using [Codespaces](https://github.com/features/codespaces). Works for Stata 15 for me. Thank you @kylebarron for making this library. Really cool!

![image](https://user-images.githubusercontent.com/39021609/130311612-5a512e39-eeda-448d-b49c-f54c6bedabf5.png)

**Comment by jiawen1998** [comment]
Hi everyone,

I encountered a very weird problem with Stata kernel using jupyter from VSCode. If I use Stata kernel and Stata cell language mode at the same time, I can't execute any commands. 
![86c49bca5b2983704ebd80b25147066](https://user-images.githubusercontent.com/90232464/176699906-0ec66374-2bfa-4fb3-abee-58908a00d0ef.png)
But if I use Stata kernel and  Python cell language mode, it works. 
![b50104b9c50f254f083fb4f0bbb705d](https://user-images.githubusercontent.com/90232464/176700301-9aeacf6b-dce9-4a08-81a0-24a73edd7b78.png)

Does anyone have any idea why it could happen? 

Other information:
1. stata version 15
![4e959ad55ef2a996bb21eed89e4e4c0](https://user-images.githubusercontent.com/90232464/176701184-e2f93194-82fc-4db3-b07c-df55b20957b7.png)

**Comment by jiawen1998** [comment]
> Hi everyone,
> 
> I encountered a very weird problem with Stata kernel using jupyter from VSCode. If I use Stata kernel and Stata cell language mode at the same time, I can't execute any commands. ![86c49bca5b2983704ebd80b25147066](https://user-images.githubusercontent.com/90232464/176699906-0ec66374-2bfa-4fb3-abee-58908a00d0ef.png) But if I use Stata kernel and Python cell language mode, it works. ![b50104b9c50f254f083fb4f0bbb705d](https://user-images.githubusercontent.com/90232464/176700301-9aeacf6b-dce9-4a08-81a0-24a73edd7b78.png)
> 
> Does anyone have any idea why it could happen?
> 
> Other information:
> 
> 1. stata version 15
>    ![4e959ad55ef2a996bb21eed89e4e4c0](https://user-images.githubusercontent.com/90232464/176701184-e2f93194-82fc-4db3-b07c-df55b20957b7.png)

Also, I find that variables are not appropriately displayed.

**Comment by kylebarron** [comment]
> But if I use Stata kernel and Python cell language mode, it works.

To me, that says that VSCode still might not really support Jupyter kernels in languages other than Python. Under the hood, Stata Kernel uses Python with a special "entry script". So when you're looking at a Python file, VSCode probably suggests all environments that run using a Python executable.

Not sure how this works in VSCode, but in Atom there had to be an agreement between the syntax highlighting package and the code running package on the name of the language. I.e. maybe there's something in either the jupyter package or the stat syntax highlighting package in VSCode that needs to change to tell vscode "_this_ language can be used with _this_ jupyter kernel".

**Comment by kylebarron** [comment]
> but I think since the stata kernel is technically installed within your python environment, you need to have python chosen to actually run the stata

This _shouldn't_ really be the case. It's true that the stata kernel runs on top of Python, but it presents itself to Jupyter as an entirely different language. So it's probably VSCode not understanding that Stata the syntax highlighter and Stata the jupyter kernel are the same and should be used together.

**Comment by kylebarron** [comment]
> does not display variables in the sidebar

I have no idea how that works in vscode, and it's certainly not implemented in Stata kernel. So an empty variable display is probably expected

**Comment by jiawen1998** [comment]
> 

Thank you for your clarification! I tried again this morning and find that it works with Stata kernel and Stata language this time. My friends also have similar problems with using stata kernel and stata language mode. Sometimes it works and other times it doesn't and must use stata kernel and python language mode.
